onChanged callback object content clarification (#31227)
mdn/content: files/en-us/mozilla/add-ons/webextensions/api/storage/storagearea/onchanged/index.md

{{AddonSidebar()}}

> Fires when one or more items in a storage area change, returning details for the keys that changed. Compared to {{WebExtAPIRef("storage.onChanged")}}, this event enables you to listen for changes in one of the storage areas: `local`, `managed`, `session`, and `sync`.

Reviewer: So, Firefox can pass changes for multiple areas, but everyone else only does one at a time? Do I understand this correctly?

Author: @jpmedley, as I understand it, the difference is that Firefox returns all the keys from the storage area, whereas other browsers only return the changed ones only.

> > **Note:** In Firefox, the information returned includes all keys within the storage area. Also, receiving a callback is possible when there is no change to the underlying data. Details of the changed items are found by examining each returned key's {{WebExtAPIRef('storage.StorageChange')}} object. See [Firefox bug 1833153](https://bugzil.la/1833153).

Reviewer: ```suggestion
> **Note:** In Firefox, the information returned includes all keys within the storage area. Also, the callback may be invoked when there is no change to the underlying data. Details of the changed items are found by examining each returned key's {{WebExtAPIRef('storage.StorageChange')}} object. See [Firefox bug 1833153](https://bugzil.la/1833153).
```
